langchain/src/retrievers/tests/vectara.int.test.ts

```diff
@@ -0,0 +1,148 @@
+    in Sub-Saharan Africa and parts of Asia, where they roam grasslands and open plains in prides consisting of related 
+    females, their offspring, and a coalition of males. Lions are known for their cooperative hunting strategy and deep 
+    social bonds, marking them as unique among the world's wild cats.`;
+
+  const flyingSquirrel = `The flying squirrel is a fascinating mammal known for its ability to glide through the air. 
+    These nocturnal creatures are found in a variety of habitats, including forests and suburban areas across parts of 
+    North America, Europe, and Asia. They possess a unique patagium, a membrane that extends from wrist to ankle, 
+    facilitating their gliding abilities. Despite their name, flying squirrels don't technically fly but instead extend 
+    their arms and legs to glide from tree to tree, using their tail for steering and balance.`;
+
+  const penguin = `Penguins are flightless seabirds known for their distinct tuxedo-like appearance and waddling walk. 
+    They are primarily found in the Southern Hemisphere, particularly in Antarctica, but species can also be found in 
+    South Africa, South America, New Zealand, and the Galapagos Islands. Penguins are expert swimmers, using their 
+    flippers for propulsion and their feet as rudders, and they can dive deep into the ocean to catch their diet of fish, 
+    squid, and krill.`;
+
+  const turtle = `Turtles are cold-blooded reptiles identifiable by their bony or cartilaginous shell developed from their 
+    ribs. They are one of the oldest and most primitive groups of reptiles and have been on Earth for more than 200 million 
+    years. Found in various habitats from deserts to marine environments around the globe, turtles are known for their 
+    slow movement on land, longevity, and often complex mating and nesting behaviors.`;
+
+  const documents = [
+    new Document({
+      pageContent: lion,
+      metadata: {
+        document_id: "1",
+        title: "Lion",
+        category: "Mammal",
+        can_fly: false,
+      },
+    }),
+    new Document({
+      pageContent: flyingSquirrel,
+      metadata: {
+        document_id: "2",
+        title: "Flying Squirrel",
+        category: "Mammal",
+        can_fly: true,
+      },
+    }),
+    new Document({
+      pageContent: penguin,
+      metadata: {
+        document_id: "3",
+        title: "Penguin",
+        category: "Bird",
+        can_fly: false,
+      },
+    }),
+    new Document({
+      pageContent: turtle,
+      metadata: {
+        document_id: "4",
+        title: "Turtle",
+        category: "Reptile",
+        can_fly: false,
+      },
+    }),
+  ];
+  return documents;
+};
+
+beforeAll(async () => {
+  // Upload the documents to Vectara
+  const store = new VectaraStore({
+    customerId: VECTARA_CUSTOMER_ID,
+    corpusId: VECTARA_CORPUS_ID,
+    apiKey: VECTARA_API_KEY,
+  });
+
+  await store.addDocuments(getdoc());
```

Comment:
You should probably mention in the documentation above that the corpus you generate should be empty from any other documents or text, so that this is going to surely work. If someone uses an existing corpus that was used with some other data, you may get a different response.